axelor-account/src/main/java/com/axelor/apps/account/service/payment/paymentvoucher/PaymentVoucherConfirmService.java

```diff
@@ -468,19 +472,16 @@ public void createMoveAndConfirm(
       // Create move line for the payment amount
       MoveLine moveLine = null;
 
+      BigDecimal currencyRate = move.getMoveLineList().get(0).getCurrencyRate();
+
       // cancelling the moveLine (excess payment) by creating the balance of all the
       // payments
       // on the same account as the moveLine (excess payment)
       // in the else case we create a classical balance on the bank account of the
       // payment mode
       BigDecimal companyPaidAmount =
-          move.getMoveLineList().stream()
-              .map(it -> isDebitToPay ? it.getCredit() : it.getDebit())
-              .reduce(BigDecimal::add)
-              .orElse(paymentVoucher.getPaidAmount());
-      companyPaidAmount = companyPaidAmount.subtract(financialDiscountAmount);
```

Comment:
You don't take the financialDiscountAmount anymore, so the payment Move amount is wrong now